buildSrc/src/main/groovy/org/elasticsearch/gradle/precommit/ThirdPartyAuditTask.groovy

```diff
@@ -40,17 +43,52 @@ import java.util.regex.Pattern;
  * Basic static checking to keep tabs on third party JARs
  */
 public class ThirdPartyAuditTask extends AntTask {
-    
+
     // patterns for classes to exclude, because we understand their issues
-    private String[] excludes = new String[0];
-    
+    private List<String> excludes = [];
+
+    @Input
+    Configuration configuration;
+
+    @InputFiles
+    FileCollection jars;
+
+    /**
+     * We use a simple "marker" file that we touch when the task succeeds
+     * as the task output. This is compared against the modified time of the
+     * inputs (ie the jars/class files).
+     */
+    @OutputFile
+    File successMarker = new File(project.buildDir, 'markers/thirdPartyAudit')
+
     ThirdPartyAuditTask() {
         // we depend on this because its the only reliable configuration
         // this probably makes the build slower: gradle you suck here when it comes to configurations, you pay the price.
         dependsOn(project.configurations.testCompile);
         description = "Checks third party JAR bytecode for missing classes, use of internal APIs, and other horrors'";
+
+        configuration = project.configurations.findByName('runtime');
+        if (configuration == null) {
+            // some projects apparently do not have 'runtime'? what a nice inconsistency,
+            // basically only serves to waste time in build logic!
+            configuration = project.configurations.findByName('testCompile');
+        }
+        assert configuration != null;
+
+        // we only want third party dependencies.
+        jars = configuration.fileCollection({ dependency ->
+            dependency.group.startsWith("org.elasticsearch") == false
```

Comment:
Are you sure this doesn't cause dependencies to be resolved during configuration?